fprime-sensors/Bmp280/Components/BmpManager/BmpManager.cpp

```diff
@@ -0,0 +1,513 @@
+                    // printf("[BmpManager] ERROR: Chip ID mismatch. Expected 0x%02X, got 0x%02X. Resetting.\n", CHIP_ID_VALUE, chip_id);
+                    this->m_state = RESET;
+                    m_resetAttempts = 0;
+                }
+            } else {
+                // printf("[BmpManager] ERROR: Failed to read chip ID. Resetting.\n");
+                this->m_state = RESET;
+                m_resetAttempts = 0;
+            }
+            break;
+        }
+        case CALIBRATION_READ:
+            //  printf("[BmpManager] CALIBRATION_READ state\n");
+            if (this->read_calibration_data()) {
+                // printf("[BmpManager] Calibration data read successfully, transitioning to CONFIGURE\n");
+                this->m_state = CONFIGURE;
+            } else {
+                // printf("[BmpManager] ERROR: Failed to read calibration data. Resetting.\n");
+                this->m_state = RESET;
+                m_resetAttempts = 0;
+            }
+            break;
+        case CONFIGURE:
+            // printf("[BmpManager] CONFIGURE state\n");
+            if (this->configure_device()) {
+                // printf("[BmpManager] Device configured successfully, transitioning to RUNNING\n");
+                this->m_state = RUNNING;
+            } else {
+                // printf("[BmpManager] ERROR: Failed to configure device. Resetting.\n");
+                this->m_state = RESET;
+                m_resetAttempts = 0;
+            }
+            break;
+        case RUNNING: {
+            // printf("[BmpManager] RUNNING state\n");
+            
+            // Step 1: Check if measurement is ready
+            U8 status = 0;
+            if (!this->read_status(status)) {
+                // printf("[BmpManager] ERROR: Failed to read status register. Resetting.\n");
+                this->m_state = RESET;
+                m_resetAttempts = 0;
+                break;
+            }
+            
+            // Check if measurement is in progress (bit 3) or if data is being updated (bit 0)
+            if ((status & 0x08) || (status & 0x01)) {
+                // printf("[BmpManager] Measurement in progress (status=0x%02X), waiting...\n", status);
+                break; // Wait for next cycle
+            }
+            
+            // Step 2: Trigger a new measurement in forced mode
+            if (!this->trigger_measurement()) {
+                // printf("[BmpManager] ERROR: Failed to trigger measurement. Resetting.\n");
+                this->m_state = RESET;
+                m_resetAttempts = 0;
+                break;
+            }
+            
+            // Step 3: Read measurement data
+            // BMP280 SPI protocol: read 6 bytes starting from PRESSURE_MSB_REGISTER
+            U8 spiData[MEASUREMENT_DATA_LENGTH + 1];
+            spiData[0] = PRESSURE_MSB_REGISTER | 0x80; // Register address with MSB=1 for read
+            for (int i = 1; i <= MEASUREMENT_DATA_LENGTH; i++) {
```

Comment:
Use of `int` raw type is forbidden.  Use a fixed-with or configurable width type.